tests/softreboot_test.go

```diff
@@ -0,0 +1,144 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	expect "github.com/google/goexpect"
+
+	v1 "kubevirt.io/client-go/apis/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	virtctlpause "kubevirt.io/kubevirt/pkg/virtctl/pause"
+	virtctlsoftreboot "kubevirt.io/kubevirt/pkg/virtctl/softreboot"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+func WaitForFedoraVMIRebooted(vmi *v1.VirtualMachineInstance) {
+	By(fmt.Sprintf("Waiting for vmi %s rebooted", vmi.Name))
+	if vmi.Namespace == "" {
+		vmi.Namespace = util.NamespaceTestDefault
+	}
+	t := time.NewTimer(time.Second * time.Duration(30))
+	defer t.Stop()
+	select {
+	case <-t.C:
```

Comment:
This can be simplified by avoiding the timer+select clause and just using `time.Sleep(30sec)` and then `Except(console.SafeExpectBatch(...))`. Not too much of a big deal though.